app/scripts/modules/core/src/pipeline/config/PipelineRegistry.ts

```diff
@@ -319,10 +317,10 @@ export class PipelineRegistry {
   // Some stages (RunJob, ?) are only setting the cloudProvider field in stage.context.
   private static resolveCloudProvider(stage: IStage): string {
     return (
-      stage.cloudProvider ||
-      stage.cloudProviderType ||
-      get(stage, ['context', 'cloudProvider']) ||
-      get(stage, ['context', 'cloudProviderType']) ||
+      stage.cloudProvider ??
+      stage.cloudProviderType ??
+      stage.context?.cloudProvider ??
+      stage?.context?.cloudProviderType ??
```

Comment:
```suggestion
      stage.context?.cloudProviderType ??
```

stage can't be nullish at this point and the difference between the lines made me look thrice